File: backend/src/main/java/org/ilisi/backend/model/Institut.java
package org.ilisi.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Institut {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private String id;

    private String name;
    private String website;
    private String imageUri;

    @OneToMany(mappedBy = "institut", cascade = CascadeType.ALL)
    private List<Review> reviews;
}

File: frontend/src/app/models/institut.ts
class Institut {
    id?: string;
    name: string;
    website: string;
    image?: File;
  imageUri?: string;

  constructor(id: string, name: string, website: string, image?: File, imageUri?: string) {
        this.id = id;
        this.name = name;
        this.website = website;
        this.image = image;
    this.imageUri = imageUri;
    }
}

export default Institut;
